### ISSUE TITLE
Not (marked as) compatible with Thunderbird 68

### ISSUE THREAD

**Issue by ojaksch** [tipichris/TBDialOut#10]
Alas, hopefully that you can make it compatible again (or set it "compatible to v68" if it's still working)

**Comment by cyberbeat** [comment]
It would be great to have this extension für thunderbird>60

**Comment by frank-eb** [comment]
Yes, because there is no alternative addon it would be really great if tbdialout is adapted to TB 68+

**Comment by McMichaeli** [comment]
Stuck as of today since Ubuntu have pushed TB 68.2.1, which TBDialOut does not support.  Anything I can do to help?

**Comment by EduardoG26** [comment]
Hallo.
I updated TB to 68.3.1 today.
TB Dialout is marked as not compatible.
Help!
Thank you. Bets regards.
Eduardo

**Comment by dantefff** [comment]
Another one that faces this issue

**Comment by frank-eb** [comment]
Seems that tipichris is offline since a longer Time.
The sourcecode is available at github and i have looked into that,
but i have no idea how to convert it to TB68+
Maybe someone who is familiar with addon development could take over
this projekt?

**Comment by tipichris** [comment]
Not offline, just busy. And for that reason it will likely be some time before I can look at this—I'm afraid paid work takes priority :). If someone wants to submit a patch I'll happily review it and incorporate it.

**Comment by dantefff** [comment]
I have created a quick and dirty pull request. In my end the extension works but the configuration interface is quite ugly.
https://github.com/dantefff/TBDialOut

**Comment by frank-eb** [comment]
Hi, danteff, works so far for initiating phonecalls. You made my day.
However configuration does not work at all. Can't see any configuration values.
But this will help till tipichris or you or someone else fix it at all.
There seem to be another change in addon-api whithin the next major releases anyway.
Its a nightmare with this.

**Comment by EduardoG26** [comment]
Brilliant, thanks danteff!
Since I'm not a programmer, how can I compile and integrate this in TB/windows7? I do need an *.xpi, right?

**Comment by tipichris** [comment]
It's not going to be that easy I'm afraid. From https://developer.thunderbird.net/add-ons/tb68/changes

> All preference related XUL elements have been removed…

The whole options.xul will need completely re-writing. There's other stuff too. Stringbundles have gone, which means all the translations need redoing. Retrieving passwords has changed, which will certainly affect the Asterisk functionality. And there is probably more.

And it all changes again for 78, so I'm starting to doubt that it's worth the effort TBH. It's annoying. TBDialout has been stable since 2012, with only one minor change to accomodate an API change for TB v. 60. Now it needs a major rewrite :(

**Comment by frank-eb** [comment]
Hi tipichris, 
maybe it would be sufficient for now to describe the about:config params and set the xpi online.
I build it from source which might not be feasable for anyone.
Totaly DAU users won't use TBDialout anyway.
For my installation the 'callto:'Settings in registry is the most important setting which i have to figure out anyway.
Unfortunately there is no alternate solution for TBDialout.
Greets from germany.

**Comment by EduardoG26** [comment]
Hallo.
I would be very grateful for a link to download the patched xpi.
AS far as I understand this will run as long as I do not change the parameters that were set with a former version. Right?
Thanks.
Eduardo

**Comment by frank-eb** [comment]
Hi, 
i try to upload the file here which workes for me but without any warranty.
I made a diff between the last version of tipichris and this one from danteff and i found nothing unexpected.
And yes, it workes with the last settings. I guess you may make changes iin the about:config if you know what you do. Hope that helps.
[tbdialout.zip](https://github.com/tipichris/TBDialOut/files/4087717/tbdialout.zip)

**Comment by EduardoG26** [comment]
Thanks. Does not work unfortunalely. The configuration seems to be broken.
![Unbenannt](https://user-images.githubusercontent.com/2843842/72817075-79263080-3c69-11ea-8d2e-e9870cf58ea3.PNG)

I have found this. Do you think It could work as a replacement sending URL-strings to my desktop Yealink?

**Comment by EduardoG26** [comment]
Funny. It works now. Many thanks!
The first time I use from TB it I need to enter user name and passwort for my VOIP phone. The user defined URL is/was saved.
Any hint?
(I think the link to voipdial, see above, is rather old!)

**Comment by EduardoG26** [comment]
This seems to have similar capability. (- open a web page with the phone number as a parameter)
https://launchpad.net/click2dial-firefox-extension
Not active for years. Source code available.

**Comment by frank-eb** [comment]
Hi Eduardo
in my case i use 'callto:'
and registry entries like:
Windows Registry Editor Version 5.00
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\callto]
@="URL:callto Protocol"
"URL Protocol"=""
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\callto\shell]
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\callto\shell\open]
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\callto\shell\open\command]
@="C:\\Program Files (x86)\\Alcatel_PIMphony\\aocphone.exe %NUM%"

![grafik](https://user-images.githubusercontent.com/57709411/72840825-5f9ade00-3c95-11ea-8683-ef0875c7eae0.png)

This one uses the cti-Tool of our telephone system there is no need for authentication because this is done in PIMphony itsef.

**Comment by marcheschi** [comment]
Hi  thanks for your work

Hi I compiled xpi, and installed it.
But I tried to fill the preferences but I cannot save them, How is possible to configure it?
Thank  you
Paolo

**Comment by frank-eb** [comment]
Hi
Well, the file prefs.js is an editable file.
Maybe you can add the values manually. But it's a guess. Do'nt know if this will work.
This are the settings contained in my file:
----
user_pref("extensions.tbdialout.passmigrated", true);
user_pref("extensions.tbdialout.plus", "00");
user_pref("extensions.tbdialout.tbbuttonadded", true);
user_pref("extensions.tbdialout.updateshown", "1.7.3");
----
This seem to be the params which are changed from default.
Good luck.

**Comment by tipichris** [comment]
I finally got some time to work on this over the weekend. So far:

- Updated the preferences dialogue to cope with the [loss of the preferences xul element](https://developer.thunderbird.net/add-ons/updating/tb68/changes#less-than-prefwindow-greater-than-less-than-prefpane-greater-than-less-than-preferences-greater-than-and-less-than-preference-greater-than)

- Changes to password storage affecting passwords for the custom URL and Asterisk AMI options, caused by [changes to the Login Manager](https://developer.thunderbird.net/add-ons/updating/tb68/changes#loginmanager) (I suspect this was the problem @EduardoG26 had)

- Worked around [changes to the stream listener](https://developer.thunderbird.net/add-ons/updating/tb68/changes#nsistreamlistener) affecting the Asterisk AMI code

I haven't (yet?) addressed the [changes to stringbundles](https://developer.thunderbird.net/add-ons/updating/tb68/changes#less-than-stringbundleset-greater-than-and-less-than-stringbundle-greater-than) because it doesn't actually seem to have happened. For example, in @EduardoG26's screenshot the title "TBDialOut Warnung" I think must have come from the stringbundle. This is actually difficult to test because on the whole the stringbundle is only used for error messages, and mostly for errors that the UI won't let you do anyway.

A little more testing and I'll post a beta xpi for you to test in the next couple of days.

**Comment by EduardoG26** [comment]
Brilliant, so many thanks!
If I can do anything like testing for this **stringbundle** thing (No idea what that is ;). ) just let me know.
Regards.

**Comment by EduardoG26** [comment]
For additional translations to Spanish and German, anytime you want...

**Comment by ojaksch** [comment]
Great! I highly apprecate this. Looking further in testing the upcoming beta.

**Comment by tipichris** [comment]
OK guys, here it is. 1.8.0pre1 should now be compatible with TB 68. There are no functional changes, just making things work with 68.

https://www.oak-wood.co.uk/downloads/tbdialout/tbdialout-1.8.0pre1.xpi

Please test and let me know about any problems. Unless one of you finds a major issue I'll aim to release this through https://addons.thunderbird.net/ as 1.8.0 shortly.

It is going to take a lot more work to make things compatible with the forthcoming release of 78, expected next month. I'll need to give some thought as to whether I can justify the time this will take. I see a lot of addon developers have moved to a paid model, but I suspect TBDialOut is too niche to actually generate any income.

**Comment by CacheMinimal** [comment]
Thanks for your work on this Chris.  I have tested 1.8.0pre1 with Ubuntu 18.04 running Thunderbird 68.7.0 on an AMD64 architecture.  I changed nothing at all in the Thunderbird set-up other than to install the new add-on over the previously marked incompatible 1.7.3, which was still present.

Everything works just fine with no configuration changes needed at all.  It once again talks to Asterisk on my FreePBX box via AMI and dials numbers from the address book.  All the UI appears to work without fault in British English.  Cheers!

**Comment by ojaksch** [comment]
Just tested a call on an internal asterisk system, which has been successful.
Will test this with a productive system tommorrow with one instance and two more instances at upcoming monday.

**Comment by EduardoG26** [comment]
Tested ok. w/ TB 68.8.0 (32-Bit) on i7 WIN7/64 pro. Many thanks!

**Comment by EduardoG26** [comment]
Regarding income. 
Why not try a "suggested payment" message as in TB for example. Di this before (!) including support for TB78, so we may pay upfront. Many of us would do because we all haved used TBDO for years already and we hope we can do so many more years.
Just generate the new version toinclude a payment message while updating.

**Comment by tipichris** [comment]
Hi @EduardoG26, 
there has been a suggested contribution for TBDO for many years. Between 2012 and 2016 I had contributions totalling $55.30, and more than half of that was from a single contribution! I don't think I've received anything since 2016. The fact that the old Mozilla contribution system has disappeared may not have helped with that, but the voluntary model isn't really paying for the time I'm afraid.

An update page should have opened in a TB tab the first time you opened the addressbook after updating. If it didn't, either you have set the `extensions.tbdialout.showupdatenotes` preference to false using about:config, or it's a bug. I'd be interested to know if this is working - it is here, but it did require some changes to get it to work again. Anyway, on that update page is a section on 'Support TBDialOut', with a freshly updated contribution link :)

**Comment by ojaksch** [comment]
> Between 2012 and 2016 I had contributions totalling $55.30

Alas, that's less than nothing!

> An update page should have opened in a TB tab the first time you opened the addressbook after updating.

It does, but TL;DR, I was horny to test it :) I'd donate some bucks, but I have no paypal account, nor will I get one. What about Patreon? Or an Amazon coupon/wishlist?

**Comment by EduardoG26** [comment]
I made a start with paypal. :).

**Comment by EduardoG26** [comment]
@ojaksch . How about asking somebody else to send it for you through paypal?

**Comment by MrMerlino** [comment]
Thanks so mutch, works correctly (TB 68.7.0 32bit @ win10pro).
donation made, thanks again!

**Comment by tipichris** [comment]
@EduardoG26 Very much appreciated, thank you.
@ojaksch I don't think you actually need a Paypal account, hence the 'Donate with a Card' button. I will look at alternatives, but I need to keep it manageable for myself too.
@MrMerlino thanks!

**Comment by ojaksch** [comment]
> @ojaksch I don't think you actually need a Paypal account, hence the 'Donate with a Card' button. I will look at alternatives, but I need to keep it manageable for myself too.

Ah, yes, I see, but that'd be handled by PayPal again. I'd choose one, but not pestilence and cholera at once ;)

**Comment by tipichris** [comment]
@ojaksch Revolut?

**Comment by ojaksch** [comment]
@tipichris Never heard of and seems to be blocked by my PiHole. What about per wire/SEPA?

**Comment by EduardoG26** [comment]
Maybe there is a nice way to place a hint on the update page that if there is enough sponsoring you will port TBDO to TB78.

**Comment by tipichris** [comment]
@ojaksch never used SEPA, being UK based. But apparently I can so I'm happy to give it a try. They tell me you need these details, and payment must be in Euro:

Beneficiary: CHRISTOPHER HASTIE
IBAN: GB78 REVO 0099 7071 7200 84
BIC: REVOGB21
Bank / Payment institution: Revolut

**Comment by ojaksch** [comment]
@tipichris Yeah, SEPA is an "EU-Thing", erm, what UK is leaving. But I'm glad that I could send my donation this way. Have a look at your bank account the next days and I sure you'll  be happier than mentioned in https://github.com/tipichris/TBDialOut/issues/10#issuecomment-624596395 :)
And thanks again for your work!

**Comment by tangram67** [comment]
Does anyboby use the latest version under Ubuntu 18.04 and TB 68.7.0?
Sounds silly, but it looks like the dialer connect button is missing.

**Comment by Tobida** [comment]
Ubuntu 18.0.4.4 LTs here, with TB 68.7.0. 
Just to be shure. You are talking about the call-button with the green handsedsymbol, to start a call out of the addressbook?
It is visible at the moment i click on a address with a phone number.

**Comment by tangram67** [comment]
Yes, I'm missing the green handset icon. There is a grayed inactive entry in the context menu called "Connect". I translated it back from German, maybe it is something called "Dial"...

![grafik](https://user-images.githubusercontent.com/2556049/81470740-ba2c5b80-91ec-11ea-8e4d-fd3fc1818b89.png)

**Comment by tipichris** [comment]
@tangram67 In the context menu you should have Rufe Privat-Telefon, Rufe Dienstlich-Telefon, and Rufe Mobil-Telefon. They should be greyed out unless you have selected a single addressbook entry and that entry has numbers for appropriate phone type. 

If the button is missing try adding it by customising your toolbar. TBDialOut does try to add a single button by default, but it is not always possible. Sometimes you will have to do this yourself. http://www.oak-wood.co.uk/oss/tbdialout#buttons

I see you have your address book in a tab. As far as I know this is not default behaviour for TB. Are you using another extension to achieve this? If so, it may be that the other extension is breaking things.

**Comment by tipichris** [comment]
@ojaksch Thanks, much appreciated

**Comment by Tobida** [comment]
I'm german too. Between "Bearbeiten | Verfassen" and "Löschen" i do have "Anrufen". 
I don't have the tab-view you have, here "Adressbuch" is a separate Window. 
Rightclicking an an entry leads to the context menu described by Chris: Rufe Privat-Telefon, Rufe Dienstlich-Telefon, and Rufe Mobil-Telefon. 
You also dont have "Verfassen".

**Comment by ojaksch** [comment]
All is working with my TB and it's own addressbook. Context menu displays all available dialing numbers, which are greyed out if no entry is selected (Arch Linux/TB 68.7.0).
No context menu in Windows 10/TB 68.8.0), but dialing works.

@tangram67 is using the **cardbook** extension, a very good alternative addressbook for TB, syncing it's content with CARDDAV servers (which TB's addressbook isn't capable of). But cardbook is using a simple **callto:** handler, sadly.
Should I open a feature request for the latter? In TBDialOut od cardbook?

**Comment by tangram67** [comment]
Looks like Chris and Larry are right. The cardbook CardDAV extension seems to break the callto handling. If I click on a phone number the default callto: handler popup is shown. I did not mention this before. Even if I try a contact from a local address book, I get the same result.

**Comment by tipichris** [comment]
As I understand it, CardBook is a completely separate address book. TBDialOut works with TB's native address book. It will not work with replacements. If you want to have both the functionality of CardBook and the functionality of TBDialOut you have two options

1. Ask the developers of CardBook to add the extra functions you want. I see their blurb refers to TBDialOut, so I guess they're familiar with it.

2. Stick with TB's native Address Book and find another way to get what CardBook gives you. Personally I use the SOGo Connector for CardDAV, although the fact that it's now rolled up in the SOGo intergrator, which has to be customised, may make it less attractive if you don't actually use SOGo. Alternatively, there is the TbSync and Provider for CalDAV & CardDAV combination. Haven't used it, but the reviews on ATN are almost as good as TBDialOut's :)

**Comment by ojaksch** [comment]
Thanks for the hint about tbsync, I'll have a deep look into this. Meanwhile I'll open an issue at cardbook to integrate/link to TBDialOut. I am sure that if they agree, there will be winners on every side.

**Comment by tangram67** [comment]
Thank you for your support.
I found the the following statement mentioned by Chris in the feature list of CardBook:
- Call phone numbers using softphone, like TBDialOut
So I will wait for the response to the CardBook issue list.

**Comment by Tobida** [comment]
If you use cardbook only because of the carddav sync: i am using TbSync with Provider for Caldav 7 Carddav. Works fine.

**Comment by ojaksch** [comment]
@Tobida As I said, Blindfisch ;) I'll have a deep lokk into it. But my goal is to have a KISS (keep it simple, stupid). No more effort than neccessary.

**Comment by tangram67** [comment]
@Tobida I will have a look at TBSync, thanks for the hint. I use CardBook with Radicale as CardDAV and CalDAV server, but that should work with TBSync as before...

**Comment by tipichris** [comment]
> No context menu nor button in Windows 10/TB 68.8.0), but dialing works.

@ojaksch Larry, to be clear, are you saying that it is not working fully as expected on Win 10/TB 68.8.0? 

I don't have a Windows installation of TB68 so haven't tested it, but I wouldn't expect it to be different. There are no platform dependant parts of TBDialOut, AFAIK. I may have to upgrade a windows installation to test, if the sun stops shining and tempting me outside :)

**Comment by ojaksch** [comment]
@tipichris Correct. I did a fresh installation of TBDialOut at a Win10 VM (playground for customers, reachable via VPN and RDP only), and while TBDialOut itself **is** working, there're no context menu or button. Let the picture speak:
![win10](https://user-images.githubusercontent.com/17717333/81491685-61fb6500-9291-11ea-9d7f-79427693bd7e.png)
I can click on one of the phone numbers and TBDialOut / Asterisk is doing it's work bravely.

**PS: Snap!** I just discovered that this VM is an uptodate Win8! Alas, I'm totally far beyond the Windows era since 2003 :) Will have a look to a local Win10 instance with libvirt/qemu ...

**Comment by ojaksch** [comment]
Okay, I could start and use a customers Win10 (as it's sunday and I'm authorized to do this). This is the beta test I did on friday, but customer is using cardbook. No dial icon per default installation, I'd to place it by menu item View/Toolbars/Customize. It's working then as normal.

**Comment by tipichris** [comment]
I've done some tests in Win10 as well as Ubuntu, and it all seems to be working. Unless I'm misreading something there don't seem to be show stopping bugs coming up here. Essentially, the new version is compatible with TB68, which was the original issue. So in the next day or so I will release 1.8.0, submit it to ATN and mark this issue closed.

There has been a bit of stuff about CardBook here, which some seem to prefer for its CardDAV capabilities. I had a look at this the other day, so I thought I'd share what I found out about TBDialOut and other extensions offering CardDAV

**CardBook**
TBDialOut is an extension that adds functionality to the Thunderbird Addressbook. CardBook is _not_ the Thunderbird Addressbook, but rather replaces it. TBDialOut does not work with it. It likely never will. Writing an extension to extend an extension is inevitably going to be fragile, unless that extension has a clear, robust, consistent and above all well documented API. 
CardBook does include a subset of TBDialOut's functionality (it will turn a phone number into a callto: link). Perhaps the authors of CardBook will extend that. But TBDialOut will not be adding support for CardBook.

**TbSync with Provider for CalDAV & CardDAV**
These are two extensions which, together, add CardDAV functionality to the TB Addressbook. The native TB Addressbook remains the source of address records. TBDialOut largely works with TbSync, but with some issues.

1. TbSync changes the way phone numbers are displayed. In doing so, it removes the part of the addressbook that TBDialOut keys into to create the clickable links. So these don't work. However, the button and the context menu are unaffected.

2. The native TB Addressbook supports three phone numbers, one each of home, work and mobile (cell). CardDAV has a much more flexible approach and can have many more numbers in a card. TbSync attempts to reconcile this problem by storing multiple numbers in a single field, separated by commas. It all works nicely when the only thing reading from that field is TbSync, which can explode the numbers and display them to the user properly. When TBDialOut tries to read this, it just sees a single number with a comma in the middle. One of the things that TBDialOut does is to strip any punctuation and formatting, assuming a phone number should be just a number. So it strips the comma. 
As long as you only have one number of each type TBDialOut will work fine. If you have, say, two work numbers then TBDialOut will in effect concatenate them and try to dial the concatenated string. I may try to resolve this in a future release, probably by just dialling the first. But for now the priority is compatibility with the next version of TB.

3. I did have the button disappear at one point. I haven't investigated why. It was easy enough to add back by customising the toolbar, but I do feel it shouldn't have disappeared. This is most likely a bug in TbSync

**SOGo Connector**
This is what I've been using for CardDAV for some years. It works perfectly with TBDialOut. Partly this is because it doesn't try to overcome the three number restriction. If you have a card with two work numbers, it will only display one of them. 

SOGo Connector is written to work with SOGo, but can work with other Card and CalDAV services. Full integration with SOGo once also needed SOGo Integrator, which need some customisation. For 68 the two have been rolled into one. But I've done a quick test and it seems although Connector does need some installation specific customisation for full integration with SOGo, it can be run 'as is' for basic DAV support, as Connector always did. It's available from https://sogo.nu/download.html#/frontends.

Hope that's of some help.

**Comment by tangram67** [comment]
Hi Chris. Of course your in depth explanation is helpful. I do fully understand that the dependencies between 2 plugins can't be solved for the reasons you explained. So for me it looks like SOGo Connector is the way to go when using CalDAV and CardDAV together with TBDialOut. I will have a look at this solution. It looks interesting. Thank you for your clarification.

**Comment by tipichris** [comment]
I have submitted v.1.8.0 to addons.thunderbird.net for review. It should be available from ATN shortly. In the meantime, it can be found at https://www.oak-wood.co.uk/downloads/tbdialout/tbdialout-1.8.0.xpi

Thanks for all your patience waiting for this. And a huge thanks to all those who have donated in the last few days ([donations are still being gratefully received](https://www.oak-wood.co.uk/oss//tbdialout/support-tbdialout) :slightly_smiling_face:)

I'll close this issue now, as TBDialOut is now compatible with Thunderbird 68. If you find further problems, please do open a new issue.

**Comment by tipichris** [comment]
For anyone who is still following this issue and interested in compatibility with Thunderbird 78, I've just put a prerelease version with 78 compatibility at https://www.oak-wood.co.uk/downloads/tbdialout/tbdialout-2.0.0pre1.xpi. Any comments or problems please add to issue #16

**Comment by ojaksch** [comment]
Chris, I'm still here and am glad to see a progress here. Will try to test this at home today and at customers site tomorrow, respectively the next days.

**Comment by ovari** [comment]
@cardbook are you able to extend your CardBook add-on to enable CardBook to work with TBDialOut? Thank you

**Comment by EduardoG26** [comment]
Hallo Chris.

I upgraded TB to 78.5.0 and your Add-On to 2.0.0pre1.

Two things:

  * I cannot dial from the address book.
  * I do not find the menu item for the settings of the ADD-ONs anymore.

Regards.

Eduardo


Am 29.11.2020 um 12:07 schrieb Chris Hastie:

**Comment by EduardoG26** [comment]
Not true. I sure can dial using your ADD-ON.

The phone numbers are not highlighted any more. I have to dial using the
right mous click menu.

Sorry.


Hallo Chris.

I upgraded TB to 78.5.0 and your Add-On to 2.0.0pre1.

Two things:

  * I cannot dial from the address book.
  * I do not find the menu item for the settings of the ADD-ONs anymore.

Regards.

Eduardo


Am 29.11.2020 um 12:07 schrieb Chris Hastie:

**Comment by tipichris** [comment]
@EduardoG26 Thanks. This issue has been discussed this morning under issue #16 (https://github.com/tipichris/TBDialOut/issues/16#issuecomment-735636369). It relates to localisation and will be corrected soon. If you find any other issues it would help if you could report them under #16

**Comment by EduardoG26** [comment]
Thank you. I will do so.

**Comment by EduardoG26** [comment]
Hallo Chris.

Are you planning an update for TB-Dialout so it would run with the 
actual Thunderbird 91.4.1?

Regards.

Eduardo

------ Originalnachricht ------
Von: "Eduardo de Mier" ***@***.***>
An: "tipichris/TBDialOut" 
***@***.***>
Gesendet: 30.11.2020 13:33:12
Betreff: Re: [tipichris/TBDialOut] Not (marked as) compatible with 
Thunderbird 68 (#10)

>Thank you. I will do so.
>
>>Am 30.11.2020 um 13:21 schrieb Chris Hastie 
***@***.***>:
>>
>>
***@***.*** <https://github.com/EduardoG26> Thanks. This issue has
